orchestration/src/main/java/com/sap/ai/sdk/orchestration/OrchestrationModuleConfig.java

```diff
@@ -211,4 +220,95 @@ public OrchestrationModuleConfig withGrounding(
       @Nonnull final GroundingProvider groundingProvider) {
     return this.withGroundingConfig(groundingProvider.createConfig());
   }
+
+  /**
+   * Creates a new configuration with the given template configuration.
+   *
+   * @link <a href="https://help.sap.com/docs/sap-ai-core/sap-ai-core-service-guide/templating">SAP
+   *     AI Core: Orchestration - Templating</a>
+   * @param templateConfig The template configuration to use.
+   * @return A new configuration with the given template configuration.
+   */
+  @Tolerate
+  @Nonnull
+  public OrchestrationModuleConfig withTemplateConfig(
+      @Nullable final TemplatingModuleConfig templateConfig) {
+
+    //    If new templateConfig is a TemplateRef, use it.
+    if (templateConfig instanceof TemplateRef) {
+      return new OrchestrationModuleConfig(
+          llmConfig, templateConfig, maskingConfig, filteringConfig, groundingConfig);
+    }
+
+    //  Make sure old responseFormat is only overwritten if new templateConfig has one set.
+    var newTemplate = (Template) templateConfig;
+    TemplateResponseFormat responseFormat = null;
+    if (this.templateConfig instanceof Template oldTemplate) {
+      responseFormat = oldTemplate.getResponseFormat();
+    }
+    //  Template.getResponseFormat() might return null, so the following check is necessary.
+    if (newTemplate != null && newTemplate.getResponseFormat() == null) {
+      newTemplate.setResponseFormat(responseFormat);
+    }
+    if (newTemplate == null && responseFormat != null) {
+      newTemplate = Template.create().template(List.of());
+      newTemplate.setResponseFormat(responseFormat);
```

Comment:
(Question)
Why not..?
```suggestion
      newTemplate.setResponseFormat(null);
```

Does the request behavior differently?

Reply:
`null` is allowed right now to be backward compatible.

Reply:
Tried to refactor the code to make it more readible.